payment/controllers.go

```diff
@@ -738,6 +773,24 @@ func VerifyCredential(service *Service) handlers.AppHandler {
 			return handlers.WrapError(err, "Error in request validation", http.StatusBadRequest)
 		}
 
+		merchant, err := GetMerchant(r.Context())
+		if err != nil {
+			return handlers.WrapError(err, "Error getting auth merchant", http.StatusInternalServerError)
+		}
+		caveats := GetCaveats(r.Context())
+
+		if req.MerchantID != merchant {
+			return handlers.WrapError(nil, "Verify request merchant does not match authentication", http.StatusForbidden)
```

Comment:
At the risk of sounding a little timid or overcautious, should we consider more generic errors here (and for the sky error below)? I have always believed in handling auth errors as generically as possible.